File: backend/src/main/java/com/ll/TeamProject/domain/user/entity/Friendship.java
package com.ll.TeamProject.domain.user.entity;

import com.ll.TeamProject.global.jpa.entity.BaseTime;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.Getter;
import lombok.Setter;

@Entity
@Getter
@Setter
public class Friendship extends BaseTime {
    // BaseTime : id (BaseEntity, no setter), 생성/수정일

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private SiteUser user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "friend_id", nullable = false)
    private SiteUser friend;

    private Status status;

    public enum Status {
        REQUESTED,
        APPROVED,
        BLOCKED
    }
}

File: backend/src/main/kotlin/com/ll/TeamProject/domain/calendar/entity/Friendship.kt
package com.ll.TeamProject.domain.friend.entity

import com.ll.TeamProject.domain.user.entity.SiteUser
import jakarta.persistence.*

@Entity
@Table(
    name = "friendship",
    uniqueConstraints = [UniqueConstraint(columnNames = ["user1_id", "user2_id"])] // 중복 저장 방지
)
data class Friendship(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,  // JPA가 자동 생성

    @ManyToOne
    @JoinColumn(name = "user1_id", nullable = false)
    val user1: SiteUser,

    @ManyToOne
    @JoinColumn(name = "user2_id", nullable = false)
    val user2: SiteUser
) {
    companion object {
        //중복된 userA, userB 들어가는거 방지
        fun create(userA: SiteUser, userB: SiteUser): Friendship {
            return if (requireNotNull(userA.id) < requireNotNull(userB.id)) {
                Friendship(user1 = userA, user2 = userB)
            } else {
                Friendship(user1 = userB, user2 = userA)
            }
        }
    }
}
